chalk-ir/src/lib.rs

```diff
@@ -1325,6 +1343,10 @@ impl<I: Interner> Goal<I> {
         Self { interned }
     }
 
+    pub fn interned(&self) -> &I::InternedGoal {
```

Comment:
I don't object to these methods, but we're not using them anywhere, right? And I don't really think they're *needed*

Reply:
Well, I was using them in rustc for debug_* functions. Alternatively, we could pass `I::Interned*` to debug_* functions